renderdoc/driver/gl/gl_driver.cpp

```diff
@@ -2190,41 +2190,31 @@ WrappedOpenGL::BackbufferImage *WrappedOpenGL::SaveBackbufferImage()
     GL.glPixelStorei(eGL_PACK_ALIGNMENT, prevPackAlignment);
 
     // scale down if necessary using simple point sampling
-    uint16_t resample_width = RDCMIN(maxSize, thwidth);
-    resample_width &= ~3;    // JPEG encoder gives shear distortion if width is not divisible by 4.
-    if(thwidth != resample_width)
-    {
-      float widthf = float(thwidth);
-      float heightf = float(thheight);
-
-      float aspect = widthf / heightf;
-
-      // clamp dimensions to a width of resample_width
-      thwidth = resample_width;
-      thheight = uint16_t(float(thwidth) / aspect);
+    thwidth = RDCMIN(maxSize, uint16_t(dat.initParams.width));
+    // TODO(akharlamov): why does OpenGL align to 4 when Vulkan and DX11 align to 8?
+    thwidth &= ~0x3;    // align down to multiple of 4
+    thheight = uint16_t(thwidth * uint16_t(dat.initParams.height / dat.initParams.width));
```

Comment:
Isn't this bracketing going to cause problems? with the `height / width` done as integers before multiplying the clamped/aligned width.

Reply:
Done.